[GSoC 2025 - Acceptance Tests] M2.6 (Fixes #23289 and part of #22716) - Implement Acceptance tests for Topic Manager

jayam04 opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR 
    - Fixes part of #22716.
    - Fixes #23289 by updating the test to use full page when comparing snapshots.
3. This PR does the following:
    - Implements acceptance tests for Topic Manager (TM) from [CUJ v3 Doc](https://docs.google.com/document/d/1D7kkFTzg3rxUe3QJ_iPlnxUzBFNElmRkmAWss00nFno/edit?tab=t.0).

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [ ] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [ ] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [ ] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [ ] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Testing doc (for PRs with Beam jobs that modify production server data)

<!--
If this PR affects production server data, please follow
[these instructions](https://github.com/oppia/oppia/wiki/Testing-jobs-and-other-features-on-production#submitting-a-pr-with-a-new-job-or-feature-that-requires-third-party-api)
and link to the job request doc here.

Otherwise, please delete this section.
-->

- [ ] A testing doc has been written: ... (ADD LINK) ...
- [ ] _(To be confirmed by the server admin)_ All jobs in this PR have been verified on a live server, and the PR is safe to merge.

## Proof that changes are correct

Proof for Fixes #23289
- https://github.com/jayam04/oppia/actions/runs/18070595391/job/51419988365 I ran test 10 times in mobile and desktop each. Two test failed but due to different error.

<!--
Add before-and-after videos/screenshots of the user-facing interface (including
the browser devtools console) to demonstrate that the changes made in this PR
work correctly. Make sure the actions taken in the before and after videos are
the same. (For changes involving responsiveness or adjustment of UI elements,
this should be a video that gradually resizes the viewport from wide to narrow
and back again.) If this PR is for a developer-facing feature, provide
videos/screenshots of the developer-facing interface instead.

When you make updates to the PR, please update these videos/screenshots as well.
You can drop videos/screenshots from previous versions of the PR.

The above should be done for all PRs, including short ones (e.g. a single-line change).
However, if the changes in your PRs are autogenerated via a script and you cannot
provide proof for the changes then please leave a comment "No proof of changes
needed because {{Reason}}" and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

jayam04 committed:
Added first test

jayam04 committed:
Updated full_stack_tests.yml

jayam04 committed:
Fixed issues with the tests

jayam04 committed:
Revert It: Removed other tests

jayam04 committed:
Added second test.

jayam04 committed:
Added third test

jayam04 committed:
Refractord many things

jayam04 committed:
Added fourth test

jayam04 committed:
Added fifth CUJ

jayam04 committed:
Added sixth acceptance test

jayam04 committed:
Analysed partial test.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/access-dashboard-using-profile-dropdown.spec.ts:
```diff
@@ -0,0 +1,174 @@
+    );
+    await topicManager.clickOnTopicAndSkillsOptionInProfileMenu();
+    await topicManager.expectScreenshotToMatch(
+      'topicsAndSkillsDashboard',
```
Leave this comment so you can update it once the screenshots are ready.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/access-dashboard-using-profile-dropdown.spec.ts:
```diff
@@ -0,0 +1,174 @@
+    );
+  });
+
+  it('should be able to view the topic with no rights', async function () {
```
"shouldn't be able to..."

jayam04 replied:
Topic manager can view the topic with no rights, however, they can't edit the topic.

imchristie replied:
Update the doc then since it's different from the doc.

jayam04 committed:
Added last test

jayam04 committed:
Added a fix.

jayam04 committed:
few fixes

jayam04 committed:
Fixes and last changes

jayam04 committed:
Some minor fix

jayam04 committed:
Minor fixes.

jayam04 committed:
Added misconception in last CUJ

jayam04 committed:
Added final things

U8NWXD commented on .github/workflows/README.md:
just a reminder to make sure we re-add the CI workflows before merging

U8NWXD commented on core/templates/pages/topic-editor-page/modal-templates/create-new-subtopic-modal.component.html:
```diff
@@ -10,7 +10,7 @@ <h3>
           <strong>Title*</strong>
         </div>
-        <input class="form-control oppia-new-subtopic-editable-field e2e-test-new-subtopic-title-field required"
+        <input class="form-control oppia-new-subtopic-editable-field e2e-test-subtopic-title-field e2e-test-new-subtopic-title-field required"
```
why do we need 2 `e2e-test-*` classes on this element?

jayam04 replied:
Done.

U8NWXD commented on core/tests/ci-test-suite-configs/acceptance.json:
```diff
@@ -358,47 +358,47 @@
     {
       "name": "topic-manager/browse-skills-on-topics-and-skills-dashboard",
-      "module": "core/tests/puppeteer-acceptance-tests/specs/topic-manager/browse-skills-on-topics-and-skills-dashboard.spec.ts"
+      "module": "core/tests/puppeteer-acceptance-tests/specs-old/topic-manager/browse-skills-on-topics-and-skills-dashboard.spec.ts"
```
what's going on with these specs? Are we actually planning to move them to this new folder, or is this just for debugging?

jayam04 replied:
Yes, I am moving them to specs-old folder. Many of the old tests have common name with the new ones.

U8NWXD commented on core/tests/ci-test-suite-configs/acceptance.json:
```diff
@@ -555,6 +555,34 @@
+    },
+    {
+      "name": "topic-manager/create-delete-and-edit-the-stories-and-chapters",
+      "module": "core/tests/puppeteer-acceptance-tests/specs/topic-manager/create-delete-and-edit-the-stories-and-chapters.spec.ts"
```
missing `core/tests/puppeteer-acceptance-tests/specs/topic-manager/create-and-edit-questions-in-skill-dashboard.spec.ts`?

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/common/user-factory.ts:
```diff
@@ -209,7 +209,6 @@ export class UserFactory {
       QuestionSubmitterFactory(),
       TopicManagerFactory(),
       CurriculumAdminFactory(),
-      VoiceoverSubmitterFactory(),
```
do we not need this for testing voiceover CUJs?

jayam04 replied:
Yes, added back.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -6622,6 +6622,87 @@ export class ExplorationEditor extends BaseUser {
+    index: number = 0
+  ): Promise<void> {
+    const selector = `${responseModalBodySelector} ${inputType}`;
+    await this.expectElementToBeVisible(selector);
```
does this only wait for one element matching `selector`? If so, we should separately wait for `element` below in case they load at different times

jayam04 replied:
I think all of the elements load at the same time, so we won't require waitfor function for each of the element.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -6622,6 +6622,87 @@ export class ExplorationEditor extends BaseUser {
+    await element.click({clickCount: 3});
+    await this.page.keyboard.press('Backspace');
+
+    await element.type(value);
```
why not use `this.type`? If it's possible to use it, I think it's preferable since it'll include any anti-flake protections we add in the future

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -6622,6 +6622,87 @@ export class ExplorationEditor extends BaseUser {
+    const element = elements[index];
+
+    // Clear all text from the element.
+    await element.click({clickCount: 3});
```
can you use `this.click` here, updating it to support multiple clicks if needed? That way we'll automatically wait for the element to load

jayam04 replied:
Currently, created a new function `clickOnElement()` in the other PR specific to clickOn function, I will update `clickOn` to use `clickOnElement`, so every clickOn functions are synced.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -6622,6 +6622,87 @@ export class ExplorationEditor extends BaseUser {
+    if (elements.length < index + 1) {
+      throw new Error(`Element ${index} not found.`);
+    }
+    await elements[index].click();
```
use `this.click`

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/exploration-editor.ts:
```diff
@@ -6622,6 +6622,87 @@ export class ExplorationEditor extends BaseUser {
+   */
+  async updateResponse(
+    index: number,
+    newAnswer?: string,
+    responseFeedback?: string
```
under what circumstances would these be undefined? seems like this function wouldn't be very useful without them

jayam04 replied:
Done. Updated args to be required.

U8NWXD reviewed (changes requested):
here's the start of my review

U8NWXD commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/access-dashboard-using-profile-dropdown.spec.ts:
```diff
@@ -0,0 +1,174 @@
+      [ROLES.TOPIC_MANAGER],
+      'Arithmetic Operations'
+    );
+  }, 600000);
```
why doesn't the default work here? ditto for the other cases of this

For reference, here's where the default is defined

https://github.com/oppia/oppia/blob/0f7d947f7dab3c9001147f95e40f2d9923d67137/core/tests/puppeteer-acceptance-tests/utilities/common/test-constants.ts#L256

jayam04 replied:
It's because the setup takes longer than default timeout of 5 minutes.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/access-dashboard-using-profile-dropdown.spec.ts:
```diff
@@ -0,0 +1,174 @@
+    );
+  }, 600000);
+
+  it('should be able to access topic and skills dashboard using via profile menu', async function () {
```
```suggestion
  it('should be able to access topic and skills dashboard using the profile menu', async function () {
```

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -257,14 +259,127 @@ const removeSkillModalHeaderSelector =
+  '.e2e-test-topics-and-skills-dashboard-link';
+const topicAndSkillsDashboardPageSelector =
+  '.e2e-test-topic-and-skills-dashboard';
+const navbarBreadcrumbSelector = 'oppia-navbar-breadcrumb';
```
can we use an `e2e-test-*` class here?

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -257,14 +259,127 @@ const removeSkillModalHeaderSelector =
+const submitSolutionButton = 'button.e2e-test-submit-solution-button';
+const saveQuestionButton = 'button.e2e-test-save-question-button';
+
+// Topic Editor > Preview Tab.
```
This comment isn't super clear. How about "Preview tab of the topic editor" or similar instead?

ditto elsewhere

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -257,14 +259,127 @@ const removeSkillModalHeaderSelector =
+   */
+  async resetTopicFilter(): Promise<void> {
+    await this.expectElementToBeVisible(resetTopicFilterButtonSelector);
+    await this.clickOn(resetTopicFilterButtonSelector);
```
can you add a post-check here?

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -512,11 +629,121 @@ export class TopicManager extends BaseUser {
+    }, interactionNameDiv);
+
+    await this.clickOn(saveInteractionButton);
+    await this.page.waitForSelector('oppia-add-answer-group-modal-component', {
```
why not define this selector at the top of the file with the others? Also, can one of the form `.e2e-test-*` be used?

jayam04 replied:
Updated from 
```
this.page.waitForSelector('oppia-add-answer-group-modal-component', {
      visible: true,
    });
``` 
to `await this.expectModalTitleToBe('Add Response');`

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -1193,16 +1466,111 @@ export class TopicManager extends BaseUser {
+    skillName: string,
+    visible: boolean = true
+  ): Promise<ElementHandle | null> {
+    const skillVisible = await this.isElementVisible(skillItem);
```
start with a pre-check to make sure the page has loaded? Otherwise a "missing" skill might just not have loaded yet

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -1193,16 +1466,111 @@ export class TopicManager extends BaseUser {
+      );
+    } else {
+      showMessage(
+        `Skill ${skillName} is visible in the skill selection modal.`
```
```suggestion
        `Skill ${skillName} is not visible in the skill selection modal.`
```

I think?

jayam04 replied:
Yes. Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -1193,16 +1466,111 @@ export class TopicManager extends BaseUser {
+      throw new Error('Radio inner circle selector not found');
+    }
+
+    await this.page.evaluate(selector => {
+      document.querySelector(selector).click();
+    }, radioInnerCircleSelector);
```
why can't we just use `this.clickOn()` here?

jayam04 replied:
I copied it from the other function.
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -1193,16 +1466,111 @@ export class TopicManager extends BaseUser {
-    await this.type(skillNameInputSelector, skillName);
+    await this.fillSkillNameInSkillSelectionModal(skillName);
 
+    // TODO: replace this with new function, once checked that new function works.
```
just leaving a note to make sure we remove this before merging

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -3168,6 +3559,394 @@ export class TopicManager extends BaseUser {
+    if (!tabElement) {
+      throw new Error(`Tab ${tabName} not found.`);
+    }
+    await tabElement.click();
```
can we use `this.clickOn()`?

jayam04 replied:
Currently, clickOn didn't worked properly with text content. However, I have updated it to use clickOn. I have also created a PR to fix the clickOn function.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -3168,6 +3559,394 @@ export class TopicManager extends BaseUser {
+
+    const questionElements = await this.page.$$(questionTextSelector);
+    let questionElement: ElementHandle<Element> | null = null;
+    for (const questionEle of questionElements) {
```
`questionEle` is a confusing variable name when we already have `questionElement`

jayam04 replied:
`questionEle` -> `questionElement`
`questionElement` -> `requiredQuestionElement`

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -3168,6 +3559,394 @@ export class TopicManager extends BaseUser {
+  async openQuestionEditor(question: string): Promise<void> {
+    const questionElement = await this.expectQuestionToBeVisible(question);
+
+    await questionElement.click();
```
use `this.clickOn()`?

jayam04 replied:
Done.

U8NWXD commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -3168,6 +3559,394 @@ export class TopicManager extends BaseUser {
+      throw new Error('Delete button not found');
+    }
+
+    await deleteButton.click();
```
use `this.clickOn()`?

jayam04 replied:
Done.

U8NWXD reviewed (changes requested):
done reviewing utility files

imchristie reviewed (commented):
Left some comment as a blocking review for @U8NWXD

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/add-questions-from-topic-editor.spec.ts:
```diff
@@ -0,0 +1,299 @@
+      'Wrong Answer'
+    );
+    await topicManager.addHintToState('Test Hint 4');
+    // TODO: Verify works properly.
```
What do you mean?

jayam04 replied:
I needed to verify that solution is added properly, verified. So, removeing the todo comment.

U8NWXD reviewed (approved):
approving to unblock since @imchristie is taking over

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.6

jayam04 committed:
Addresses few comments

jayam04 committed:
Few fixes

jayam04 committed:
Added few fixes

jayam04 committed:
Fixes

jayam04 committed:
Some fixes

jayam04 committed:
Few more fixes

jayam04 committed:
Fixes some of the tests

jayam04 committed:
Fixes some of the tests (2)

jayam04 committed:
Fix

jayam04 committed:
Added debug log

jayam04 committed:
Fix

jayam04 committed:
Fix

jayam04 committed:
Added some fixes and debugging logs

jayam04 committed:
Fix

jayam04 committed:
Type fixes

jayam04 committed:
Fix

jayam04 committed:
Fixes the test

jayam04 committed:
Fixes mobile tests

jayam04 committed:
Fixes tests

jayam04 committed:
Updates full_stack_tests.yml

jayam04 committed:
Adds more logging

jayam04 committed:
Few changes

jayam04 committed:
Fix

jayam04 committed:
Fix

jayam04 committed:
Fix: add-question-from-topic-editor:183

jayam04 committed:
Fix: addDragAndDropSortSolution

jayam04 committed:
Fixes: edit-the-topic.spec.ts & add-questions-from-topic-editor.spec.ts:91

jayam04 committed:
Fix: add-questions-from-topic-editor:221 (desktop)

jayam04 committed:
Fix: Uncommented closeBrowser()

jayam04 committed:
Fix: selector

jayam04 committed:
Fix: add-questions-from-topic-editor.spec.ts:188

jayam04 committed:
Fix

jayam04 committed:
Revert: addInteraction

jayam04 committed:
Fixes

jayam04 committed:
Update: full_stack_tests.yml

jayam04 committed:
Reverts: waitForElementToBeClickable to not show logs

jayam04 committed:
Fix: Multiple buttons with same selector

jayam04 committed:
Fix: edit-and-republish-the-skill.spec.ts:112

jayam04 committed:
Fix: Desktop test

jayam04 committed:
Fix: mobile test (edit and republish the skill)

jayam04 committed:
Fix: create-delete-and-edit-the-stories-and-chapters.spec.ts (mobile)

jayam04 committed:
Fix: add-questions-from-topic-editor.spec.ts:240

jayam04 committed:
Fix: selector

jayam04 committed:
Bug fixes

jayam04 committed:
Fixes

jayam04 committed:
Fixes

jayam04 committed:
Fix: add-questions-from-topic-editor.spec.ts

jayam04 committed:
Updates full_stack_tests.yml

jayam04 committed:
Fix: selector in mobile

jayam04 committed:
Addresess few of the comments

jayam04 committed:
Updated full_stack_tests.yml

jayam04 reviewed (commented):
Addressed few comments

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.6

jayam04 committed:
Address clickOn comments

jayam04 committed:
Adds screenshots

jayam04 committed:
Minor fixes

jayam04 committed:
Merge branch 'develop' into m2.6

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -326,14 +338,57 @@ const questionDifficultySelectionModalSelector =
+    if (this.isViewportAtMobileWidth()) {
+      await this.expectElementToBeVisible(displayMobileFiltersButton);
+      await this.clickOn(displayMobileFiltersButton);
+      await this.clickOn(mobileTopicFilterResetSelector);
```
Missing post-check

jayam04 replied:
Done.

jayam04 committed:
Adds skill update warning

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -326,14 +341,57 @@ const questionDifficultySelectionModalSelector =
+      await this.clickOn(mobileTopicFilterResetSelector);
+    } else {
+      await this.expectElementToBeVisible(resetTopicFilterButtonSelector);
+      await this.clickOn(resetTopicFilterButtonSelector);
```
Missing post-check

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts:
```diff
@@ -583,11 +585,22 @@ export class BaseUser {
-    await this.page.waitForSelector(selector, {visible: true});
-    await this.waitForElementToBeClickable(selector);
-    await this.page.type(selector, text);
+  async type(
```
Rename this function to something like "typeInInputField" for clearer description.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/logged-out-user.ts:
```diff
@@ -6582,6 +6582,7 @@ export class LoggedOutUser extends BaseUser {
     await this.page.waitForFunction(
       (selector: string, expectedTabTitle: string) => {
         const tabTitle = document.querySelector(selector)?.textContent?.trim();
+        console.log('[debug] tabTitle: ', tabTitle);
```
Reminder to take this out before merging.

jayam04 replied:
Done.

imchristie commented on core/tests/puppeteer-acceptance-tests/utilities/user/topic-manager.ts:
```diff
@@ -3602,45 +3673,87 @@ export class TopicManager extends BaseUser {
 
-    const skillStatus = await skillContainer.$eval(
+        console.log(
+          '[debug]: (SkillElement) ' + skillElement?.textContent?.trim()
```
Reminder to take these out before merging.

jayam04 replied:
Done.

jayam04 replied:
Removed all.

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/create-and-edit-questions-in-skill-dashboard.spec.ts:
```diff
@@ -0,0 +1,442 @@
+    );
+
+    await topicManager.expectScreenshotToMatch(
+      'updatedQuestionInQuestionEditor',
```
Missing screenshots

jayam04 replied:
Done

imchristie reviewed (commented):
Completed full review

jayam04 committed:
Fix: [Bug] where answer was added to wrong modal and wrong answer was marked as correct.

jayam04 committed:
Fix: Added selectors, lint fixes, etc.

jayam04 committed:
Addresses comments (1)

jayam04 reviewed (commented):
Addressed comments.

jayam04 committed:
Revert "Revert It: Removed other tests"

This reverts commit 88118a5e72a49932c4b0e079781187d9cc5666c1.

jayam04 committed:
Update tests.yml

jayam04 committed:
Fix

jayam04 committed:
Fix

jayam04 committed:
Fix

jayam04 committed:
Fixes the test "Edit the test from Skill Editor"

jayam04 committed:
Updates acceptance tests to cover "should show Learn Again button
correctly" from E2E tests

jayam04 committed:
Updates acceptance tests to cover "should open appropriate modal on
re-clicking an interaction to" from E2E tests.

jayam04 committed:
Revert "Updates acceptance tests to cover "should open appropriate modal on"

This reverts commit 4a7546d2cb2354c5b29dda9cf820c7240c72cf70.

jayam04 committed:
Revert "Updates acceptance tests to cover "should show Learn Again button"

This reverts commit 288a87e3d20aa631e09a99ab2dd66c4fc79f81d1.

jayam04 committed:
Fixes.

jayam04 committed:
Fixes the test.

jayam04 committed:
Fix flake

jayam04 committed:
Adds back removed selector

jayam04 committed:
Fix: "Create Question" button blocked by the navigation

jayam04 committed:
Fixes "Create Question" blocked by navigation in mobile

jayam04 committed:
Updates test.yml

jayam04 committed:
Fixes test: edit-and-preview-a-subtopic

jayam04 committed:
Revert "Adds skill update warning"

This reverts commit 812b50f1bd9aca3e6b9d163bee32c9538ff36afb.

jayam04 committed:
Removes the unused functions

jayam04 committed:
Fixes

jayam04 committed:
Removes "waitForTransitionEnd"

jayam04 committed:
Update exploration-editor.ts

jayam04 committed:
Skip covered input field test

jayam04 committed:
Comment out legacy tests and update test matrix params

jayam04 committed:
Add parallel test workflow for faster execution

diff --git a/core/templates/pages/admin-page/admin-page.module.ts
b/core/templates/pages/admin-page/admin-page.module.ts index
c52b4abf47..1fcdccd46c 100644 ---
a/core/templates/pages/admin-page/admin-page.module.ts +++
b/core/templates/pages/admin-page/admin-page.module.ts

jayam04 committed:
Add question editor screenshot snapshots

jayam04 committed:
Remove test workflow files

imchristie commented on core/tests/puppeteer-acceptance-tests/specs/topic-manager/create-and-edit-questions-in-skill-dashboard.spec.ts:
```diff
@@ -360,58 +363,61 @@ describe('Topic Manager', function () {
-    ]) {
-      await topicManager.expectQuestionToPreviewProperly(question);
-    }
+    // TODO(): Currently, there is issue where all questions are not previewed
```
Are you planning to create an issue for this?

jayam04 replied:
Yes, done.

imchristie reviewed (commented):
Left one comment.

jayam04 committed:
Update create-subtopic-study-guide-with-multiple-sections.spec.ts

jayam04 committed:
Update create-and-edit-questions-in-skill-dashboard.spec.ts

jayam04 committed:
Fix interaction tab selection in exploration editor

The commit removes redundant tab switching code since the functionality
was already handled by changeTabInInteractionSelectionModal method.

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.6

jayam04 committed:
Fix: Lint errors

imchristie reviewed (approved):
LGTM

jayam04 commented:
Hi @Nik-09 Can you review this PR as LaCE-frontend-reviewer

Nik-09 reviewed (approved):
LGTM for codeowner files. Thanks

jayam04 committed:
Increase timeout for subtopic test to 500000ms

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.6

jayam04 committed:
Type fixes from the merge

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.6

jayam04 committed:
Lint Fixes.

jayam04 committed:
Lint Fixes

jayam04 committed:
Fix: Merge confict not resolved properly.

jayam04 committed:
Update puppeteer-utils.ts

jayam04 committed:
Fixes

jayam04 committed:
Fixes

jayam04 committed:
Fix: Response not marked as correct.

jayam04 committed:
Update curriculum-admin.ts

jayam04 committed:
Fix #23289: Update screenshot and use full screen snapshot for flaky test.

jayam04 committed:
Fix: Lint errors

jayam04 committed:
Fix

imchristie reviewed (approved):
New commits look good to me

jayam04 committed:
Merge remote-tracking branch 'upstream/develop' into m2.6
